pkg/server/server.go

```diff
@@ -272,3 +290,47 @@ func (s *Server) newEndpointsServer(catalog catalog.Catalog, svidRotator svid.Ro
 func (s *Server) caCertsPath() string {
 	return path.Join(s.config.DataDir, "certs.json")
 }
+
+func (s *Server) validateTrustDomain(ctx context.Context, ds datastore.DataStore) error {
+	trustDomain := s.config.TrustDomain.Host
+
+	fetchResponse, err := ds.ListRegistrationEntries(ctx, &datastore.ListRegistrationEntriesRequest{})
+	if err != nil {
+		s.config.Log.Error(err)
+		return errors.New("error trying to fetch entries")
```

Comment:
Why not return the error here instead of logging it and returning a different one? Or something like this

```go
return fmt.Errorf("check existing entries: %v", err)
```